src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2271,10 +2271,18 @@ internal void AddInterception(string filePath, int line, int character, Location
             var callLineColumn = callLocation.GetLineSpan().Span.Start;
             Debug.Assert(callLocation.SourceTree is not null);
             var key = (callLocation.SourceTree.FilePath, callLineColumn.Line, callLineColumn.Character);
-            if (_interceptions.TryGetValue(key, out var oneInterception))
+
+            if (_interceptions.TryGetValue(key, out var interceptionsAtAGivenLocation))
             {
-                // We don't expect to reach this phase if there are duplicate interceptors in the compilation.
-                return oneInterception.Single();
+                if (interceptionsAtAGivenLocation is [var oneInterception])
+                {
+                    return oneInterception;
+                }
+
+                // We don't normally reach this branch in batch compilation.
+                // One scenario where we may reach this is when validating used assemblies, which performs lowering of method bodies even if declaration errors would be reported.
+                // See 'CSharpCompilation.GetCompleteSetOfUsedAssemblies'.
+                diagnostics.Add(ErrorCode.ERR_ModuleEmitFailure, callLocation, this.SourceModule.Name, new LocalizableResourceString(nameof(CSharpResources.ERR_DuplicateInterceptor), CodeAnalysisResources.ResourceManager, typeof(CodeAnalysisResources)));
```

Comment:
I couldn't figure out how to hit this code path in a way which also lets us verify the diagnostic which gets reported here.

Reply:
Since we'll already have reported an error, maybe we just return null here (no interception) without reporting any additional diagnostic?